source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -31,6 +35,121 @@ def __init__(self, obj, position, _rangeObj=None):
 					1
 				)
 
+	def move(self, unit, direction, endPoint=None):
+		oldRange=None
+		if not self._isCaret:
```

Comment:
Note that every TextInfo instance has an attribute `basePosition` that should be textInfos.POSITION_CARET. Therefore, I think I'd prefer this:

```suggestion
		if self.basePosition != textInfos.POSITION_CARET:
```